client/controllers/smartEvents/smartEvent.coffee

```diff
@@ -5,15 +5,39 @@ Incidents = require '/imports/collections/incidentReports'
 Modal.allowMultiple = true
 
 Template.smartEvent.onCreated ->
-  @editState = new ReactiveVar false
+  @editState = new ReactiveVar(false)
   @eventId = new ReactiveVar()
+  @loading = new ReactiveVar(true)
+
+Template.smartEvent.onRendered ->
+  eventId = Router.current().getParams()._id
+  @eventId.set(eventId)
+  @subscribe 'smartEvents', eventId
   @autorun =>
-    eventId = Router.current().getParams()._id
-    @eventId.set eventId
-    @subscribe 'smartEvents', eventId,
-      onReady: =>
-        event = SmartEvents.findOne(eventId)
-        @subscribe 'smartEventIncidents', disease: event.disease
+    event = SmartEvents.findOne(eventId)
+    if event
+      eventDateRange = event.dateRange
+      locations = event.locations
+      disease = event.disease
+      query = {}
+      if disease
+        query.disease = disease
+      if eventDateRange
+        query['dateRange.start'] = $lte: eventDateRange.end
+        query['dateRange.end'] = $gte: eventDateRange.start
+      if locations
+        locationNames = []
+        for location in locations
+          featureCode = location.featureCode
+          locationNames.push(location.countryName)
+          if featureCode is 'ADM1' or featureCode is 'ADM2'
+            locationNames.push(location.admin1Name)
+          if featureCode is 'ADM2'
+            locationNames.push(location.admin2Name)
+        query['locations.name'] = $in: _.uniq(locationNames)
```

Comment:
This should match on the corresponding geoname properties rather than the location name. This is going to match the incident reports that mention a location containing the selected location rather than those contained by the selected location.